### PR TITLE
docs: layered rewrite — low-barrier getting-started, new CanIDeploy hub, plugin docs audit

### PR DESCRIPTION
**Pull request by sahina**
## Summary
- Layered rewrite of getting-started + guides around a single persona (junior dev told to add contract tests): TL;DR visible, walkthrough H2s, `<details>` for depth, never tabs for consumer-vs-producer.
- New `docs/guides/can-i-deploy.mdx` hub page pulls scattered deploy-safety content from breaking-changes, producer-testing, and ci-cd-integration into one canonical place; sidebar + homepage card updated.
- Plugin docs pass: close real gaps (verification steps, `cvt plugins list` sample output, plugin metrics surfaced to users, safe-mode recovery promoted, "Go-only in v1" explicit, 30-line minimum-viable-plugin skeleton); fix a stale helper name in the design doc. All factual claims cross-checked against `pkg/cvt`, `server/cvtservice`, `internal/pluginmgr`, `cmd/cvt/plugins.go`.

17 files, net −1903 lines while adding content (can-i-deploy.mdx + verification sections); savings come from deduplicating quickstart content across the guides and collapsing edge cases into `<details>`. Docusaurus build is clean — no broken anchors, no MDX errors.

## Test plan
- [ ] Browse the rendered docs site locally (`npm run start` in `docs-site/`) and read intro → quick-start → consumer-testing → producer-testing → can-i-deploy top-to-bottom as persona C.
- [ ] Confirm all four `<details>` patterns render: intro (3 blocks), quick-start (4), guides (varies), plugins/authoring-go (6).
- [ ] Verify sidebar shows `CanIDeploy` between Producer Testing and Breaking Changes.
- [ ] Homepage "Safe Deployment Checks" card links to `/docs/guides/can-i-deploy`.
- [ ] Spot-check that claimed CLI output (`cvt plugins list` table) matches real output when run locally.

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive `can-i-deploy` deployment safety guide with workflow instructions and consumer registry integration.
  * Restructured installation and quick-start guides for improved clarity and streamlined onboarding.
  * Updated consumer testing, producer testing, and breaking-changes guides with simplified examples and clearer workflows.
  * Enhanced plugin documentation with decision tables, configuration reference, and troubleshooting sections.
  * Reorganized getting-started content and related guide links across the documentation site.

### CODE REVIEW

File: docs/guides/producer-testing.mdx [hunk 1]
```diff
@@ -184,28 +90,19 @@ def test_kit():
...
-        method="GET",
-        path="/pet/123",
-        status_code=200,
-        body={"id": 123},  # missing 'name' and 'photoUrls' fields
+        method="GET", path="/pet/123",
+        status_code=200, body={"id": 123},
     )
-
     assert not result.valid
     assert "name" in result.errors[0]
```
[the remark this answers is not in the document]
Comment (sahina) [reply docs/guides/producer-testing.mdx:107]: Declining this change. Verified against the Python SDK: `ProducerValidationResult` in `sdks/python/cvt_sdk/producer/testing.py:135` is a `@dataclass` (attribute access, `.valid`). The consumer-side `ValidationResult` in `sdks/python/cvt_sdk/__init__.py:93` is a `TypedDict` (dict access, `["valid"]`). The two sides intentionally use different Python types, so each guide correctly mirrors its SDK surface. No doc change needed.

File: docs/guides/producer-testing.mdx [hunk 2]
```diff
@@ -315,417 +185,200 @@ public class PetHandlerTest {
...
-ProducerConfig config = ProducerConfig.builder()
-    .schemaId("petstore")
-    .validator(validator)
-    .mode(ValidationMode.STRICT)
-    .excludePaths(List.of("/health", "/metrics", "/ready"))
-    .includePaths(List.of("/api/**"))
-    .build();
+// Spring
+registry.addInterceptor(new SpringInterceptor(config))
+    .addPathPatterns("/api/**");
```
[the remark this answers is not in the document]
Comment (sahina) [reply docs/guides/producer-testing.mdx:299]: Fixed in 0206274. Added the `ProducerConfig.builder()` definition before `registry.addInterceptor` in the Java middleware tab for parity with the other language tabs.

File: docs/intro.md [hunk 3]
```diff
@@ -8,156 +8,91 @@ slug: /intro
...
-### Getting Started
+<details>
+<summary>New to OpenAPI? (1 minute)</summary>
 
-- **[Installation](./getting-started/installation.mdx)** - Install the server and SDKs
-- **[Quick Start](./getting-started/quick-start.mdx)** - Your first contract test
+OpenAPI is a machine-readable format for describing HTTP APIs — paths, methods, parameters, and response shapes. If your API has a Swagger UI page, there's an OpenAPI spec behind it. CVT accepts OpenAPI v3 natively and auto-converts Swagger 2.0.
 
-### Guides
+Learn more: [openapi.io](https://www.openapis.org/).
```
[the remark this answers is not in the document]
Comment (sahina) [reply docs/intro.md]: Fixed in 0206274. Link text changed from `openapi.io` to `openapis.org` to match the URL.

File: docs/plugins/authoring-go.md [hunk 4]
```diff
@@ -158,14 +135,14 @@ func (r *myRegistry) FetchSchema(ctx context.Context, req *registrypb.FetchSchem
...
-## Connection pooling
+</details>
 
-Reuse a single `http.Client` across calls. The default client opens a
-new connection per request, which is wasteful when CVT sends many
-consecutive `FetchSchema` calls.
+<details>
+<summary>Connection pooling (reuse http.Client)</summary>
+
+Reuse a single `http.Client` across calls. The default client opens a new connection per request, which is wasteful when CVT sends many consecutive `FetchSchema` calls.
```
[the remark this answers is not in the document]
Comment (sahina) [reply docs/plugins/authoring-go.md]: Fixed in 0206274. Reworded to remove the inaccurate "opens a new connection per request" claim. The real issue is throwing away the pool by constructing a new `http.Client` / `http.Transport` per call — updated accordingly, plus a note about draining and closing response bodies so keep-alive connections return to the pool.

File: docs/plugins/README.md [hunk 5]
```diff
@@ -1,93 +1,100 @@
...
-- You just want to customize a single CVT build for your org — fork or
-  vendor instead.
-- You need to swap the storage backend — that's in-tree via
-  `server/storage/factory.go`, not a plugin.
+# 4. trigger the bound hook (e.g., `cvt validate --schema some-id`) and watch
+#    cvt_plugin_call_duration_seconds_count increment for your plugin
+curl -s http://localhost:9551/metrics | grep cvt_plugin_call_duration_seconds_count
+```
+
+Four plugin metrics are always emitted: `cvt_plugin_call_duration_seconds`, `cvt_plugin_call_errors_total`, `cvt_plugin_up`, `cvt_plugin_restarts_total`.
```
[the remark this answers is not in the document]
Comment (sahina) [reply docs/plugins/README.md]: Fixed in 0206274. Verified against `internal/pluginmgr/metrics.go`: `cvt_plugin_up` is labelled by `{plugin}` only; `cvt_plugin_info{plugin,version,sha256}` is the fifth metric that carries identity labels. Updated the metrics grep, the example output, and the enumeration ("four" → "five").

File: docs/plugins/README.md [hunk 6]
```diff
@@ -97,52 +104,40 @@ the plugin author publishes.
...
 **In v1:**
 - Two extension points: `RegistryProvider` and `EventHandler`.
 - One plugin per hook (no fanout).
-- Go SDK.
+- Go SDK only.
 - Linux + macOS.
-- Three CLI commands: `cvt plugins list`, `install`, `remove`.
-- Four Prometheus metrics + synchronous audit.
+- `cvt plugins {list, install, remove}`.
+- Four plugin metrics + synchronous audit.
```
[the remark this answers is not in the document]
Comment (sahina) [reply docs/plugins/README.md]: Fixed in 0206274. Scope bullet updated to "Five plugin metrics + synchronous audit." to match the verification section and the actual implementation. Also updated `docs/design/plugin-system.md` metrics list for the same reason.

File: docs/plugins/reference-plugins.md [hunk 7]
```diff
@@ -1,73 +1,48 @@
...
-releases via the `github.com/sahina/cvt` Go module dependency.
-Each plugin's own repo README carries the full config surface and a
-quick-test recipe; this page is the walkthrough.
+| You want to… | Plugin | Contract |
+|---|---|---|
+| Fetch OpenAPI specs from a REST schema registry (Apicurio, Backstage, internal API registry) | [`cvt-plugin-rest`](https://github.com/sahina/cvt-plugin-rest) | `RegistryProvider` |
+| Track consumer→schema usage in a central system | [`cvt-plugin-rest`](https://github.com/sahina/cvt-plugin-rest) | `RegistryProvider` |
+| Post breaking-change alerts to Slack | [`cvt-plugin-slack`](https://github.com/sahina/cvt-plugin-slack) | `EventHandler` |
+| Alert the on-call channel on validation failures | [`cvt-plugin-slack`](https://github.com/sahina/cvt-plugin-slack) | `EventHandler` |
+| Something else | Write your own — [authoring guide](authoring-go.md) |
```
[the remark this answers is not in the document]
Comment (sahina) [reply docs/plugins/reference-plugins.md]: Fixed in 0206274. Added the missing third cell to the "Something else" row so the 3-column table renders cleanly.

File: docs/plugins/reference-plugins.md [hunk 8]
```diff
@@ -86,61 +61,35 @@ hooks:
...
-  pair fails validation — in CI, from `cvt validate`, or in a running
-  `cvt serve`. Includes per-plugin dedup so a misconfigured producer
-  throwing 10k failures/minute doesn't page your channel 10k times.
-- `OnBreakingChangeDetected`: formats a Slack message listing each
-  breaking change (kind, method, path) and posts to the webhook.
-  Fires from two paths: the `CompareSchemas` gRPC method (diff-tool
-  / PR-comment-bot pattern) and from `RegisterSchema` when
-  `check_compatibility=true` (CLI: `cvt register-schema --check-compatibility`).
+- `OnValidationFailed` → Slack message describing the failed interaction (method, path, schema ID, first validation error). Fires from `ValidateInteraction`. Plugin-side dedup so a misconfigured producer throwing 10k failures/minute doesn't page your channel 10k times.
+- `OnBreakingChangeDetected` → Slack message listing each breaking change (kind, method, path). Fires from `CompareSchemas` and from `RegisterSchema --check-compatibility`.
```
[the remark this answers is not in the document]
Comment (sahina) [reply docs/plugins/reference-plugins.md]: Fixed in 0206274. Reworded to: "Fires from `pkg/cvt.Validator.Validate` when it returns a non-valid result (CLI `cvt validate` path and library callers). Note: the server's `ValidateInteraction` gRPC method runs validation directly via `openapi3filter` and does not call this hook in v1." Verified against `pkg/cvt/hooks_fire.go` + `pkg/cvt/validator.go`.
